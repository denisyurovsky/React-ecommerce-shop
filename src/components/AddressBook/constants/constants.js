export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
