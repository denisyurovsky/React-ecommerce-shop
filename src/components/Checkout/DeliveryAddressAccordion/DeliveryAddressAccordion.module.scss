@import "../../../assets/styles/colors.scss";

.accordion {
  width: 100%;
  border-top-left-radius: 15px !important;
  border-top-right-radius: 15px !important;
  margin-bottom: 15px;

  &::before {
    display: none;
  }
}

.accordionSummary {
  background-color: $greyMain !important;
}

.header {
  display: flex;
  align-items: center;
}

.icon {
  width: 30px !important;
  height: 30px !important;
  margin-right: 15px !important;
  transition: 1s;
}

.subtitle {
  font-size: 18px;
}

.formControl {
  width: 100%;
}

.radioGroup {
  width: 100% !important;
}

.formControlLabel {
  margin: 0 !important;
  margin-bottom: 30px !important;

  & > *:last-child {
    flex-grow: 1;
  }
}

.button {
  margin-top: 20px !important;
}
