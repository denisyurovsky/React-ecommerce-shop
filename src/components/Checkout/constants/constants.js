export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
