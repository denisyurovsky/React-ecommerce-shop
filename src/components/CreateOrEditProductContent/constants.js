export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
