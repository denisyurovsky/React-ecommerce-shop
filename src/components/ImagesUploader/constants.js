export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
