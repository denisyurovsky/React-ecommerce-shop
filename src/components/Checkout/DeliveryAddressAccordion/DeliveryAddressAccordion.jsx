import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import EditIcon from '@mui/icons-material/Edit';
import LocalShippingIcon from '@mui/icons-material/LocalShipping';
import {
  Accordion,
  AccordionDetails,
  Box,
  Typography,
  Button,
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup,
} from '@mui/material';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { useContext } from 'react';

import { CheckoutContext } from '../../../pages/CheckoutPage/CheckoutPage';
import { PANEL } from '../constants/constants';
import AccordionSummary from '../helpers/AccordionSummary';

import AccordionSummaryInfo from './AccordionSummaryInfo/AccordionSummaryInfo';
import Label from './Label/Label';

import styles from './DeliveryAddressAccordion.module.scss';

const DeliveryAddressAccordion = ({
  isDeliveryAddressExpended,
  handleDeliveryAddressButton,
  handleChangeAddresses,
  addressId,
  handleEditButton,
  handleChangeAccordion,
  addresses,
  address,
}) => {
  const [disabledAccordion] = useContext(CheckoutContext);

  return (
    <>
      <Accordion
        className={classNames({
          [styles.accordionSummary]: !isDeliveryAddressExpended,
          [styles.accordion]: true,
        })}
        expanded={isDeliveryAddressExpended}
        onChange={handleChangeAccordion(PANEL.DELIVERY_ADDRESS)}
      >
        <AccordionSummary
          disabled={disabledAccordion.address}
          expandIcon={
            isDeliveryAddressExpended ? <ArrowDropDownIcon /> : <EditIcon />
          }
          data-testid="deliveryAddressSummery"
        >
          <Box className={styles.header}>
            <LocalShippingIcon className={styles.icon} />
            <Box>
              <Typography className={styles.subtitle} variant="subtitle2">
                Delivery address
              </Typography>
              {!isDeliveryAddressExpended && addressId !== null && (
                <AccordionSummaryInfo address={address} />
              )}
            </Box>
          </Box>
        </AccordionSummary>
        <AccordionDetails>
          <FormControl className={styles.formControl}>
            <RadioGroup
              className={styles.radioGroup}
              onChange={handleChangeAddresses}
              value={addressId}
            >
              {addresses.data.length > 0 &&
                addresses.data.map((address) => (
                  <FormControlLabel
                    className={styles.formControlLabel}
                    key={address.id}
                    value={address.id}
                    control={<Radio />}
                    label={
                      <Label
                        handleEditButton={handleEditButton}
                        address={address}
                      />
                    }
                  />
                ))}
            </RadioGroup>
          </FormControl>
          <Button
            onClick={handleDeliveryAddressButton}
            fullWidth
            className={styles.button}
            variant="contained"
            disabled={addressId === null}
          >
            Proceed
          </Button>
        </AccordionDetails>
      </Accordion>
    </>
  );
};

DeliveryAddressAccordion.propTypes = {
  handleDeliveryAddressButton: PropTypes.func.isRequired,
  handleChangeAddresses: PropTypes.func.isRequired,
  handleEditButton: PropTypes.func.isRequired,
  handleChangeAccordion: PropTypes.func.isRequired,
  addresses: PropTypes.object.isRequired,
  address: PropTypes.object.isRequired,
  isDeliveryAddressExpended: PropTypes.bool.isRequired,
  addressId: PropTypes.number,
};

export default DeliveryAddressAccordion;
