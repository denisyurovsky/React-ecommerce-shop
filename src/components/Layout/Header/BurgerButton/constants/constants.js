import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
