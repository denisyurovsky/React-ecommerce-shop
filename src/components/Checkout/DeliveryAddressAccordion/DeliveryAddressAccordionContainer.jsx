import { createBrowserHistory } from 'history';
import PropTypes from 'prop-types';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { toast } from 'react-toastify';

import { getAddressById } from '../../../api/addresses';
import { getProductsByIds } from '../../../api/products';
import { notificationError } from '../../../constants/constants';
import { pathNames } from '../../../constants/pathNames';
import getSelectedProductIds from '../../../helpers/getSelectedProductIds';
import {
  deleteSelectedProducts,
  selectCart,
} from '../../../store/cart/cartSlice';
import { addOrder, editAddressOrder } from '../../../store/orders/ordersSlice';
import { PANEL } from '../constants/constants';

import DeliveryAddressAccordion from './DeliveryAddressAccordion';

const DeliveryAddressAccordionContainer = ({
  setCreatedOrderId,
  setExpanded,
  expanded,
  handleChangeAddresses,
  addressId,
  handleEditButton,
  handleChangeAccordion,
  addresses,
  address,
  setIsPaymentMethodAccordionDisabled,
  orderId,
}) => {
  const dispatch = useDispatch();
  const cart = useSelector(selectCart);
  const isDeliveryAddressExpended = expanded === PANEL.DELIVERY_ADDRESS;

  const handleEditOrder = async () => {
    try {
      const addressResponse = await getAddressById(addressId);

      dispatch(
        editAddressOrder({
          orderId,
          addressData: addressResponse.data,
          addressId,
        })
      );
    } catch (err) {
      toast.error(notificationError);
    }
  };

  const handleCreateOrder = async () => {
    try {
      const checkedProductIds = getSelectedProductIds(cart.sellers);
      const productsResponse = await getProductsByIds(checkedProductIds);
      const addressResponse = await getAddressById(addressId);

      const res = await dispatch(
        addOrder({
          productsData: productsResponse.data,
          addressData: addressResponse.data,
          addressId,
        })
      );

      setCreatedOrderId(res.payload.id);
      createBrowserHistory().push(`${pathNames.CHECKOUT}/${res.payload.id}`);

      dispatch(deleteSelectedProducts());
    } catch (err) {
      toast.error(notificationError);
    }
  };

  const handleDeliveryAddressButton = () => {
    setIsPaymentMethodAccordionDisabled(false);
    setExpanded(PANEL.PAYMENT_METHOD);

    if (orderId) {
      handleEditOrder();
    } else {
      handleCreateOrder();
    }
  };

  return (
    <DeliveryAddressAccordion
      isDeliveryAddressExpended={isDeliveryAddressExpended}
      handleDeliveryAddressButton={handleDeliveryAddressButton}
      handleChangeAddresses={handleChangeAddresses}
      addressId={addressId}
      handleEditButton={handleEditButton}
      handleChangeAccordion={handleChangeAccordion}
      addresses={addresses}
      address={address}
    />
  );
};

DeliveryAddressAccordionContainer.propTypes = {
  setCreatedOrderId: PropTypes.func.isRequired,
  setExpanded: PropTypes.func.isRequired,
  handleChangeAddresses: PropTypes.func.isRequired,
  handleEditButton: PropTypes.func.isRequired,
  handleChangeAccordion: PropTypes.func.isRequired,
  setIsPaymentMethodAccordionDisabled: PropTypes.func.isRequired,
  addresses: PropTypes.object.isRequired,
  address: PropTypes.object.isRequired,
  expanded: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
  orderId: PropTypes.number,
  addressId: PropTypes.number,
};

export default DeliveryAddressAccordionContainer;
