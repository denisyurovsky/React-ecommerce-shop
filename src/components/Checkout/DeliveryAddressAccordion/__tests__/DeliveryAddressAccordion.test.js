import React from 'react';

import { EMPTY_ADDRESS } from '../../../../pages/AddressBookPage/constants/constants';
import { addressesDto } from '../../../../test-utils/dto/addressesDto';
import renderWith from '../../../../test-utils/renderWith';
import DeliveryAddressAccordion from '../DeliveryAddressAccordion';

describe('DeliveryAddressAccordion component', () => {
  it('should render valid snapshot', () => {
    const { asFragment } = renderWith(
      <DeliveryAddressAccordion
        handleDeliveryAddressButton={() => {}}
        handleChangeAddresses={() => {}}
        handleEditButton={() => {}}
        handleChangeAccordion={() => {}}
        isDeliveryAddressExpended={true}
        addressId={null}
        addresses={{ data: addressesDto }}
        address={EMPTY_ADDRESS}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
