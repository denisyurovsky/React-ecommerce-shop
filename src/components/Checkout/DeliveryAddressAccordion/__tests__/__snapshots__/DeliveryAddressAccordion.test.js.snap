// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddressAccordion component should render valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters accordion css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-24vt7b-MuiButtonBase-root-MuiAccordionSummary-root"
      data-testid="deliveryAddressSummery"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <div
          class="header MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="LocalShippingIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8h-3V4H3c-1.1 0-2 .9-2 2v11h2c0 1.66 1.34 3 3 3s3-1.34 3-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3h2v-5l-3-4zM6 18.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm13.5-9 1.96 2.5H17V9.5h2.5zm-1.5 9c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
            />
          </svg>
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 subtitle css-ati914-MuiTypography-root"
            >
              Delivery address
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            >
              <div
                class="MuiFormControl-root formControl css-1nrlq1o-MuiFormControl-root"
              >
                <div
                  class="MuiFormGroup-root radioGroup css-dmmspl-MuiFormGroup-root"
                  role="radiogroup"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd formControlLabel css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        name="mui-1"
                        type="radio"
                        value="0"
                      />
                      <span
                        class="css-hyxlzm"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                          data-testid="RadioButtonUncheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                          data-testid="RadioButtonCheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      <div
                        class="addressCard MuiBox-root css-0"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            Mr. Alexander Sidorov
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            +7 (981) 3002020
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph country css-1sn4lm3-MuiTypography-root"
                          >
                            Russian Federation, Saint Petersburg
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            Pulkovskaya, 8, 4
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            168142
                          </span>
                        </div>
                        <span
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium iconButton css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                          role="button"
                          tabindex="0"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="EditIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                      </div>
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd formControlLabel css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        name="mui-1"
                        type="radio"
                        value="2"
                      />
                      <span
                        class="css-hyxlzm"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                          data-testid="RadioButtonUncheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                          data-testid="RadioButtonCheckedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      <div
                        class="addressCard MuiBox-root css-0"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            Mrs. Alexandra Sidorova
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            +7 (981) 1233232
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph country css-1sn4lm3-MuiTypography-root"
                          >
                            Russian Federation, Saint Petersburg
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            Pulkovskaya, 8, 4
                          </span>
                          <span
                            class="MuiTypography-root MuiTypography-caption paragraph css-1sn4lm3-MuiTypography-root"
                          >
                            168142
                          </span>
                        </div>
                        <span
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium iconButton css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                          role="button"
                          tabindex="0"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="EditIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                      </div>
                    </span>
                  </label>
                </div>
              </div>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                Proceed
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
